tests/utils_tests/test_archive.py

```diff
@@ -42,6 +44,16 @@ def test_extract_function(self):
         extract(self.archive_path, self.tmpdir)
         self.check_files(self.tmpdir)
 
+    @unittest.skipIf(
+        sys.platform == 'win32',
+        'Python on Windows has a limited os.chmod() support.',
+    )
+    def test_extract_file_permissions(self):
+        extract(self.archive_path, self.tmpdir)
+        filepath = os.path.join(self.tmpdir, '1')
```

Comment:
Are the other files in the added archives needed? Can a more descriptive name than "foobar" be used?